src/NuGetForUnity/Editor/UnityPathHelper.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.IO;
+using UnityEngine;
+
+namespace NugetForUnity
+{
+    /// <summary>
+    ///     Helper class for Unity-Project related paths.
+    /// </summary>
+    internal static class UnityPathHelper
+    {
+        static UnityPathHelper()
+        {
+            AbsoluteAssetsPath = Path.GetFullPath(Application.dataPath);
+            AbsoluteProjectPath = Path.GetDirectoryName(AbsoluteAssetsPath) ?? throw new InvalidOperationException("Can't detect project root.");
+        }
+
+        /// <summary>
+        ///     Gets the absolute path to the Unity-Project 'Assets' directory.
+        /// </summary>
+        internal static string AbsoluteAssetsPath { get; }
+
+        /// <summary>
+        ///     Gets the absolute path to the Unity-Project root directory.
+        /// </summary>
+        internal static string AbsoluteProjectPath { get; }
+
+        /// <summary>
+        ///     Checks if given path is within Assets folder.
+        /// </summary>
+        /// <param name="path">Path to check.</param>
+        /// <returns>True if path is within Assets folder, false otherwise.</returns>
+        internal static bool IsPathInAssets(string path)
+        {
+            return Path.IsPathRooted(path) || GetAssetsRelativePath(path).StartsWith("..");
```

Comment:
Shouldn't this condition be inverted?

Reply:
Oh you are right I completely scrambled this while moving it to a separat function.